activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -45,8 +45,10 @@ def _assign_attributes(attributes)
 
       def _assign_attribute(k, v)
         setter = :"#{k}="
+        public_send(setter, v)
+      rescue NoMethodError
         if respond_to?(setter)
```

Comment:
Hi  @skipkayhil I'm curious and would like to ask: if at line 48 we call `public_send(setter, v)` and it goes inside the `NoMethodError` rescue block, why do we need to call `if respond_to?(setter)` at line 50, since we know the method does not exist?

Reply:
Hey, thanks for asking! This is needed because we don't know for sure _which_ method is throwing the `NoMethodError`. It _could_ be the `setter`, but it also could be something called by the `setter`. So using `respond_to?` here is an easy way to verify: Is this `NoMethodError` being raise by `setter` specifically or something else?

Reply:
Thanks for the explanation. Very clear! This can happen for example with custom writers. ( my mind was focused on attr_writer type of writers only haha )